### PR TITLE
🌱  Bump golangci-lint

### PR DESCRIPTION
**Pull request by benmoss**
**What this PR does / why we need it**:
- Bumps golangci-lint to 1.28.1
- Globally disable godot, goerr113. Godot seems insignificant, goerr113 is more substantial of changes than I want to make right now.
- Use specific nolints everywhere else

It'd be nice to fix the `nestif` `nolints` but not gonna tackle them now. It would also probably be nice to use static errors everywhere and enable `goerr113` but I felt like that was too significant a change.

### CODE REVIEW

File: cmd/clusterawsadm/cmd/alpha/migrate/migrate.go [hunk 1]
```diff
@@ -94,7 +95,7 @@ Supported versions: %v`, supportedVersions),
 
 	newCmd.Flags().StringVarP(&clusterName, "clusterName", "n", "", "name of existing Cluster object")
 	if err := newCmd.MarkFlagRequired("clusterName"); err != nil {
-		_ = fmt.Errorf("error making required flag")
+		panic(errors.Wrap(err, "error making required flag"))
```
Comment (benmoss) [inline cmd/clusterawsadm/cmd/alpha/migrate/migrate.go:98]: This is worth noting, we were swallowing this error before. It should never happen though and I don't think there's a way to surface it without some larger changes.

File: cmd/clusterawsadm/cmd/alpha/alpha.go [hunk 2]
```diff
@@ -17,24 +17,20 @@ limitations under the License.
...
+func AlphaCmd() *cobra.Command {
 	newCmd := &cobra.Command{
 		Use:   "alpha",
 		Short: "alpha commands",
 		Long:  `Alpha commands may not be supported in future releases`,
-		Run: func(cmd *cobra.Command, args []string) {
-			if err := cmd.Help(); err != nil {
-				_ = fmt.Errorf("error printing Help")
-			}
+		RunE: func(cmd *cobra.Command, args []string) error {
```
Comment (benmoss) [inline cmd/clusterawsadm/cmd/alpha/alpha.go:32]: This was not related to a linter but we were also swallowing this error, it's easy enough to change it to `RunE` and let cobra handle it.

**Comment by benmoss** [comment]
Hmm, not sure why the generated files are out of sync. Is this just a bug in something?

```diff
-       // WARNING: in.Bastion requires manual conversion: inconvertible types (sigs.k8s.io/cluster-api-provider-aws/api/v1alpha2.Instance vs *sigs.k8s.io/cluster-api-provider-aws/api/v1alpha3.Instance)
+       // WARNING: in.Bastion requires manual conversion: inconvertible types (./api/v1alpha2.Instance vs *sigs.k8s.io/cluster-api-provider-aws/api/v1alpha3.Instance)

-            description: 'APIVersion defines the versioned schema of this representation
-              of an object. Servers should convert recognized schemas to the latest
-              internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+            description: 'APIVersion defines the versioned schema of this representation of an object. Servers should convert recognized schemas to the
latest internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
```

**Comment by randomvariable** [comment]
@benmoss this has been intermittently happening. both me and @detiber have attempted to solve it once and for all to no avail.

The workaround right now is to check out the repo to ${GOPATH}/src/sigs.k8s.io/cluster-api-provider-aws.

File: .golangci.yml [hunk 3]
```diff
@@ -11,6 +11,8 @@ linters:
     - whitespace
     - gocognit
     - gomnd
+    - godot
+    - goerr113
```
Comment (detiber) [inline .golangci.yml:15]: Might be good to create a tracking issue for going back and cleaning up `goerr113` related issues.

File: controllers/awsmachine_controller_test.go [hunk 4]
```diff
@@ -21,6 +21,7 @@ import (
 	"context"
 	"flag"
 	"fmt"
+
 	"sigs.k8s.io/cluster-api/util/conditions"
```
Comment (detiber) [inline controllers/awsmachine_controller_test.go:26]:
```suggestion

	"sigs.k8s.io/cluster-api/util/conditions"
```

File: pkg/cloud/services/ec2/console.go [hunk 5]
```diff
@@ -18,6 +18,7 @@ package ec2
 
 import (
 	"encoding/base64"
+
 	"sigs.k8s.io/cluster-api-provider-aws/pkg/record"
```
Comment (detiber) [inline pkg/cloud/services/ec2/console.go:23]:
```suggestion

	"sigs.k8s.io/cluster-api-provider-aws/pkg/record"
```

File: pkg/cloud/services/ec2/gateways.go [hunk 6]
```diff
@@ -18,6 +18,7 @@ package ec2
 
 import (
 	"fmt"
+
 	"sigs.k8s.io/cluster-api/util/conditions"
```
Comment (detiber) [inline pkg/cloud/services/ec2/gateways.go:23]:
```suggestion

	"sigs.k8s.io/cluster-api/util/conditions"
```

File: pkg/cloud/services/ec2/routetables.go [hunk 7]
```diff
@@ -17,9 +17,10 @@ limitations under the License.
 package ec2
 
 import (
-	"sigs.k8s.io/cluster-api/util/conditions"
 	"strings"
 
+	"sigs.k8s.io/cluster-api/util/conditions"
+
```
Comment (detiber) [inline pkg/cloud/services/ec2/routetables.go:23]: This should be grouped with the below imports, and the two separate groups of imports below should likely also be grouped together

File: pkg/cloud/services/ec2/securitygroups.go [hunk 8]
```diff
@@ -18,6 +18,7 @@ package ec2
 
 import (
 	"fmt"
+
 	"sigs.k8s.io/cluster-api/util/conditions"
 
 	errlist "k8s.io/apimachinery/pkg/util/errors"
```
Comment (detiber) [inline pkg/cloud/services/ec2/securitygroups.go:24]: similar here for imports, these should be consolidated (and also consolidated with the grouping below)

File: pkg/cloud/services/ec2/subnets.go [hunk 9]
```diff
@@ -19,10 +19,11 @@ package ec2
...
 	"math/rand"
-	"sigs.k8s.io/cluster-api/util/conditions"
 	"sort"
 	"strings"
 
+	"sigs.k8s.io/cluster-api/util/conditions"
+
 	"sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/awserrors"
 	"sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/converters"
 	"sigs.k8s.io/cluster-api-provider-aws/pkg/cloud/services/wait"
```
Comment (detiber) [inline pkg/cloud/services/ec2/subnets.go:29]: similar here for import grouping

File: test/e2e/conformance_test.go [hunk 10]
```diff
@@ -33,6 +29,11 @@ import (
 	"text/template"
 	"time"
 
+	. "github.com/onsi/ginkgo"
+	"github.com/onsi/ginkgo/config"
+	. "github.com/onsi/gomega"
+	"github.com/pkg/errors"
+
```
Comment (detiber) [inline test/e2e/conformance_test.go:36]:
```suggestion

	"github.com/pkg/errors"
```

File: test/e2e_new/common.go [hunk 11]
```diff
@@ -21,9 +21,11 @@ package e2e_new
 import (
 	"context"
 	"fmt"
+	"path/filepath"
+
 	. "github.com/onsi/ginkgo"
 	corev1 "k8s.io/api/core/v1"
-	"path/filepath"
+
```
Comment (detiber) [inline test/e2e_new/common.go:28]:
```suggestion
```

**Pull request merged**